Wrappers/Python/cil/optimisation/algorithms/FISTA.py

```diff
@@ -70,58 +126,169 @@ def __init__(self, initial=None, f=None, g=ZeroFunction(), **kwargs):
             else:
                 raise ValueError('{} received both initial and the deprecated x_init parameter. It is not clear which one we should use.'\
                     .format(self.__class__.__name__))
-        
-        if initial is not None and f is not None:
-            self.set_up(initial=initial, f=f, g=g)
 
-    def set_up(self, initial, f, g=ZeroFunction()):
-        '''initialisation of the algorithm
+        self._step_size = None
+        self.set_up(initial=initial, f=f, g=g, step_size=step_size, **kwargs)
 
-        :param initial: Initial guess ( Default initial = 0)
-        :param f: Differentiable function
-        :param g: Convex function with " simple " proximal operator'''
+    def set_up(self, initial, f, g, step_size, **kwargs):
+        """ Set up of the algorithm
+        """
 
-        logging.info("{} setting up".format(self.__class__.__name__, ))
-        
-        self.y = initial.copy()
+        # set up ISTA      
+        self.initial = initial
         self.x_old = initial.copy()
-        self.x = initial.copy()
-        self.u = initial.copy()
-
+        self.x = initial.copy()           
         self.f = f
         self.g = g
-        if f.L is None:
-            raise ValueError('Error: Fidelity Function\'s Lipschitz constant is set to None')
-        self.invL = 1/f.L
-        self.t = 1
-        self.configured = True
+
+        # set step_size
+        self.set_step_size(step_size=step_size)
+
+        # check convergence criterion for ISTA is satisfied
+        if kwargs.get('check_convergence_criterion', True):
```

Comment:
why not `check_if_provably_convergent`?

Reply:
Renamed to `check_if_provably_convergent`, this should be use also in #1256.

Reply:
I think we need to discuss it again and find an optimal design for all of our algorithms.